### ISSUE TITLE
Where to send solved small bugs?

### ISSUE THREAD

**Issue by mohsen1365ir** [TomSchimansky/CustomTkinter#147]
Dear Tom and all,
First, I'd like to thank you for the precious work that you've done here. I've started changing the widgets of my program from Tkinter to customTkinter.  During so, I might find some bugs in the library that I fix. My question is where should I report the fixed bugs?
For example:
**customtkinter_button.py line 141: elif isinstance(self.master, (ttk.Frame, ttk.LabelFrame, ttk.Notebook))**
as ttk has another widget name ttk.Label, it should be added to this line:
**elif isinstance(self.master, (ttk.Frame, ttk.LabelFrame, ttk.Notebook, ttk.Label))**
Regards,
Mohsen

**Comment by TomSchimansky** [comment]
Hey, you can simply list them here, and I will fix them, or you could create a pull request. But I think the easiest way is, if you just post them here. Thanks for doing this!

**Comment by mohsen1365ir** [comment]
Thank you for the prompt answer. I will post them here if I find any.

**Comment by mohsen1365ir** [comment]
I was replacing tk.CheckButton with ctk.CTkCheckBox. I realized the command function has different behaviours at the initialization.  i.e. the command is run for CTk while it is not run for the Tk.
here is an example:

import tkinter as tk
import customtkinter as ctk

win = tk.Tk()
var1 = tk.BooleanVar(value = 0)
var2 = tk.BooleanVar(value = 0)
a = tk.Checkbutton(win, variable=var1, command = lambda: print('tk.Checkbutton'))
b = ctk.CTkCheckBox(win, variable= var2, command = lambda: print('ctk.CTkCheckBox'))
a.pack()
b.pack()
win.mainloop()

and the output:
>>> ctk.CTkCheckBox

Do you think this is normal or needs to be fixed?
Thanks

**Comment by mohsen1365ir** [comment]
> I was replacing tk.CheckButton with ctk.CTkCheckBox. I realized the command function has different behaviours at the initialization. i.e. the command is run for CTk while it is not run for the Tk. here is an example:
> 
> import tkinter as tk import customtkinter as ctk
> 
> win = tk.Tk() var1 = tk.BooleanVar(value = 0) var2 = tk.BooleanVar(value = 0) a = tk.Checkbutton(win, variable=var1, command = lambda: print('tk.Checkbutton')) b = ctk.CTkCheckBox(win, variable= var2, command = lambda: print('ctk.CTkCheckBox')) a.pack() b.pack() win.mainloop()
> 
> and the output:
> 
> > > > ctk.CTkCheckBox
> 
> Do you think this is normal or needs to be fixed? Thanks

The easiest way for me to fix this is as below:
line 95: self.check_state = (variable.get() == onvalue)
delete line 129 to 132

Basically, you don't need to call select or deselect in __init__ because you've already drawn it.  The only parameter that needs to be taken care of is self.check_state.
Cheers,

**Comment by mohsen1365ir** [comment]
I'm not sure whether my comments are helpful or not; I found another issue:
It's almost the same code that I sent, except it's printing var.get() which is the state of checkBox. [checked = 1, unchecked = 0], but it shows reverse:
import tkinter as tk
import customtkinter as ctk

win = tk.Tk()
var1 = tk.BooleanVar(value = 0)
var2 = tk.BooleanVar(value = 0)
a = tk.Checkbutton(win, variable=var1, command = lambda: print('tk.Checkbutton', var1.get()))
b = ctk.CTkCheckBox(win, variable= var2, command = lambda: print('ctk.CTkCheckBox', var2.get()))
a.pack()
b.pack()
win.mainloop()

![Untitled](https://user-images.githubusercontent.com/58194459/171975263-f1d96811-ed9f-4e6f-8aaa-74311f187e8a.jpg)

I realized in the toggle method, self.function is called before self.variable is updated: therefore it should be changed to this:

    def toggle(self, event=0):
        if self.state == tkinter.NORMAL:
            if self.check_state is True:
                self.check_state = False
                self.draw()
            else:
                self.check_state = True
                self.draw()

            if self.variable is not None:
                self.variable_callback_blocked = True
                self.variable.set(self.onvalue if self.check_state is True else self.offvalue)
                self.variable_callback_blocked = False

            if self.function is not None:
                self.function()

**Comment by mohsen1365ir** [comment]
It seems that textvariable has no usage so far.

```
import tkinter as tk
import customtkinter as ctk

win = tk.Tk()
var1 = tk.BooleanVar(value = 0)
var2 = tk.BooleanVar(value = 0)
text_var = tk.StringVar(value = 'text_var')
text_var2 = tk.StringVar(value = 'text_var')
a = tk.Checkbutton(win, variable=var1, text = 'text', textvariable= text_var, 
                    command = lambda: print('tk.Checkbutton', var1.get()))
b = ctk.CTkCheckBox(win, variable= var2, text = 'text', textvariable= text_var2, 
                    command = lambda: print('ctk.CTkCheckBox', var2.get()))
a.pack()
b.pack()
win.mainloop()
```
![image](https://user-images.githubusercontent.com/58194459/171994823-19584045-31cc-43dc-9788-ff9bcd1f0bdd.png)

These are the changes that I made:

line 87: 
```
        if textvariable is None:
            self.text = text
        else:
            self.text = textvariable.get()
```

At the end of init :
```
if self.textvariable is not None:
            self.textvariable_callback_name = self.textvariable.trace_add("write", self.textvariable_callback)
```
Added method:
```
def textvariable_callback(self, var_name, index, mode):
        self.set_text(self.textvariable.get())
```

Added at the end of configure method:
```
if "textvariable" in kwargs:
            if self.textvariable is not None:
                self.textvariable.trace_remove("write", self.textvariable_callback_name)

            self.textvariable = kwargs["textvariable"]

            if self.textvariable is not None and self.textvariable != "":
                self.textvariable_callback_name = self.textvariable.trace_add("write", self.textvariable_callback)
                self.set_text(self.textvariable.get())
            else:
                self.textvariable = None

            del kwargs["textvariable"]
```
result:
![image](https://user-images.githubusercontent.com/58194459/171994948-97b4152e-8d05-4a10-977e-ab078885d699.png)

**Comment by mohsen1365ir** [comment]
customtkinter_radiobutton.py:

When disabled, the command function still works (when clicked over it) while in the tk.Radiobutton the command won't work.

changes:
self.function inside invoke method should be called only if self.state = 'normal':

```
    def invoke(self, event=0):
        if self.state == tkinter.NORMAL:
            if self.function is not None:
                self.function()
            if self.check_state is False:
                self.check_state = True
                self.select()
```

Thanks

**Comment by mohsen1365ir** [comment]
I realized my version of customtkinter was a bit old and all of the names have changed. I'll reconsider these and let you know. Meanwhile I appreciate it if you take a look and fix them(if needed)

**Comment by mohsen1365ir** [comment]
Hi. Any update? @TomSchimansky

**Comment by mohsen1365ir** [comment]
I can create a pull request with all of the changes so far if you don't mind? (changes to the latest version)

**Comment by yevheniia1smortsova** [comment]
Hi,

It appears the CTkSlider widget has the same feature: it executes the command while initializing ... 

all the best,
Yevheniia

**Comment by TomSchimansky** [comment]
I fixed this for the slider now with version 4.4.0, I will have a look at the checkbox and the pull request too.
